File: gradle_boot_server/src/main/java/net/canway/component/JerseyConfig.java
package net.canway.component;

import org.glassfish.jersey.jackson.JacksonFeature;
import org.glassfish.jersey.server.ResourceConfig;
import org.glassfish.jersey.server.spring.SpringComponentProvider;
import org.glassfish.jersey.server.spring.scope.RequestContextFilter;
import org.springframework.stereotype.Component;

import javax.ws.rs.ApplicationPath;


/**
 * 配置jersay
 */
@Component
@ApplicationPath("rest")
public class JerseyConfig extends ResourceConfig {

    public JerseyConfig() {
        register(RequestContextFilter.class);
        packages("net.canway.controller");
        register(JacksonFeature.class);
        register(RequestContextFilter.class);
//        register(LoggingFilter.class);
        register(SpringComponentProvider.class);
    }
}

File: gradle_boot_client/src/main/java/net/canway/component/JerseyConfig.kt
package net.canway.component

import org.glassfish.jersey.jackson.JacksonFeature
import org.glassfish.jersey.server.ResourceConfig
import org.glassfish.jersey.server.spring.SpringComponentProvider
import org.glassfish.jersey.server.spring.scope.RequestContextFilter
import org.springframework.stereotype.Component

import javax.ws.rs.ApplicationPath


/**
 * 配置jersay
 */
@Component
@ApplicationPath("rest")
class JerseyConfig : ResourceConfig() {
    init {
        register(RequestContextFilter::class.java)
        packages("net.canway.controller")
        register(JacksonFeature::class.java)
        register(RequestContextFilter::class.java)
        //        register(LoggingFilter.class);
        register(SpringComponentProvider::class.java)
    }
}
